File: 백준/Gold/14921. 용액 합성하기/용액 합성하기.py
n = int(input())

arr = list(map(int, input().split()))

ans = float('inf')
left = 0
right = n - 1

while left < right:
    diff = arr[right] + arr[left]

    if abs(diff) < abs(ans):
        ans = diff

    if diff < 0:
        left += 1
    else:
        right -= 1

print(ans)

File: 백준/Gold/14921. 용액 합성하기/용액 합성하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        int[] arr = new int[n];

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int ans = Integer.MAX_VALUE;

        int left = 0;
        int right = n - 1;

        while (left < right) {

            int diff = arr[right] + arr[left];

            if (Math.abs(diff) < Math.abs(ans)) ans = diff;

            if (diff == 0) break;

            if (diff < 0) left++;
            else right--;
        }

        System.out.println(ans);
    }
}